### DISCUSSION TITLE
VAD inference time

### DISCUSSION THREAD

**Discussion by kuruvachankgeorge** [pyannote/pyannote-audio#949] [Q&A]
Hi folks! I'm wondering if anyone has compared the inference time (latency) of the Pyannote VAD with other popular VAD algorithms, like Silero-VAD etc. I could find a comparison on their performance in terms of False alarm and Missed detection rates in the [paper](https://arxiv.org/pdf/2104.04045.pdf), but nothing is clearly mentioned about the latency or computational load. Would be great if you can share these details as well. May I also know if pyannote segmentation model supports onnx or tensorRT (for reducing the inference time) because my intention is to integrate the VAD to my ASR engine (by replacing the webrtc VAD for better speech-silence segmentation) for real-time inferencing. Thanks!

**Comment by hbredin** [comment]
Thanks for your interest in pyannote. 

Improving inference time is definitely not the priority right now as pyannote.audio was not initially designed for real-time processing.

That being said, 
* [this blog post](https://herve.niderb.fr/fastpages/2021/08/05/Streaming-voice-activity-detection-with-pyannote.html) may help you setting up streaming VAD with pyannote.
* [this paper](https://github.com/juanmc2005/StreamingSpeakerDiarization/blob/main/paper.pdf) provides some numbers (though it does more than just VAD)

Feel free to contribute a proper (independent) benchmark.

**Reply by kuruvachankgeorge** [reply]
Thanks @hbredin
